app/services/sources/source.ts

```diff
@@ -51,6 +54,30 @@ export class Source implements ISourceApi {
     return this.getObsInput().settings;
   }
 
+  /**
+   * Compares the details of this source to another, to determine
+   * whether adding as a reference makes sense.
+   * @param comparison the comparison details of the other source
+   */
+  isSameType(comparison: ISourceComparison): boolean {
+    if (this.channel) return;
```

Comment:
Since we expose this function to public API the returning type should be consistent with the method definition. Because we don't know how other clients parse the API responses. So should be `return false` here instead `return`